File: Team C (Season 2)/Week4/Chaekie/신고결과받기.py
def solution1(id_list, report, k):
    answer = []
    reporter_dict = {id: set() for id in id_list}
    attacker_dict = {id: set() for id in id_list}
    result_dict = {id: 0 for id in id_list}

    for case in report:
        reporter = case.split()[0]
        attacker = case.split()[1]
        reporter_dict[reporter].add(attacker) 
        attacker_dict[attacker].add(reporter) 

    for attacker in attacker_dict:
        if len(attacker_dict[attacker]) >= k:
            for reporter, attackers in reporter_dict.items():
                if attacker in attackers:
                    result_dict[reporter] += 1

    answer = list(result_dict.values())

    return answer



# report 자체를 set 해주면 됨
# id_list의 인덱스값을 이용해 reporter 이름이랑 매치함
def solution2(id_list, report, k):
    answer = [0] * len(id_list)    
    report_dict = {id : 0 for id in id_list}

    for reporter in set(report):
        report_dict[reporter.split()[1]] += 1

    for reporter in set(report):
        if report_dict[reporter.split()[1]] >= k:
            answer[id_list.index(reporter.split()[0])] += 1

    return answer

File: Team C (Season 2)/Week 4/Cliff/신고결과받기.swift
func receiveReport(_ id_list: [String], _ report: [String], _ k: Int) -> [Int] {
    var reporterDict: [String: [String]] = [:]
    var reporteeDict: [String: Int] = [:]
    
    for reportPair in report {
        let splitted = reportPair.split(separator: " ")
        let reporter = String(splitted[0])
        let reportee = String(splitted[1])
        
        // 신고 한 리스트: 한 유저가 같은 유저를 여러 번 신고한 경우는 신고 횟수 1회로 처리
        if reporterDict[reporter] == nil {
            reporterDict[reporter, default: []].append(reportee)
            reporteeDict[reportee, default: 0] += 1
        } else if let reportees = reporterDict[reporter], !reportees.contains(reportee) {
            reporterDict[reporter, default: []].append(reportee)
            reporteeDict[reportee, default: 0] += 1
        } // 신고 당한 리스트: 한 사람이 여러번 신고하였다면 카운트하지 않음
    }
    
    // 메일 전송 횟수
    var result: [Int] = [Int](repeating: 0, count: id_list.count)
    for (index, id) in id_list.enumerated() {
        guard let reportees = reporterDict[id] else {
            continue
        }
        
        for reportee in reportees {
            if reporteeDict[reportee]! >= k {
                result[index] += 1
            }
        }
    }
    
    return result
}

receiveReport(["muzi", "frodo", "apeach", "neo"], ["muzi frodo","apeach frodo","frodo neo","muzi neo","apeach muzi"], 2)
receiveReport(["con", "ryan"], ["ryan con", "ryan con", "ryan con", "ryan con"], 3)
